core/src/main/java/org/jruby/RubyFileTest.java

```diff
@@ -444,7 +444,7 @@ public static RubyBoolean writable_p(IRubyObject recv, IRubyObject filename) {
             return RubyFileTest.writable_p(recv, filename);
         }
 
-        @JRubyMethod(name = "zero?", required = 1)
+        @JRubyMethod(name = {"empty?", "zero?"}, required = 1)
         public static RubyBoolean zero_p(ThreadContext context, IRubyObject recv, IRubyObject filename) {
             return RubyFileTest.zero_p(context, recv, filename);
```

Comment:
@kcdragon shoot.  I messed up!  I thought your deifnition of zero_p was on RubyFileTest and not on RubyPathname.  MRI does do a dynamic call (callMethod or rb_funcall in MRI) in pathname.  sorry for messing that up.  You had it right the first time :|